src/main/java/seedu/address/logic/commands/ImportVolunteerCsvCommand.java

```diff
@@ -0,0 +1,128 @@
+
+    private static final String MESSAGE_IMPORT_COMPLETED = "Volunteer(s) imported from CSV file "
+            + "to your Desktop.";
+    private static final String MESSAGE_IMPORT_VOLUNTEER_FAILED = "Volunteer(s) import failed, please try again.";
+    private static final String MESSAGE_IMPORT_VOLUNTEER_LACK_INFO = "Volunteer information incomplete.\n"
+            + "Please check format and try again";
+
+
+    private final FileReader csvFile;
+
+    /**
+     * @param file to be read as a csv data
+     */
+    public ImportVolunteerCsvCommand(FileReader file) {
+        csvFile = file;
+    }
+
+    @Override
+    public CommandResult execute(Model model, CommandHistory history) throws CommandException {
+        BufferedReader br = new BufferedReader(csvFile);
+        try {
+            // this is the first line and it should be the title row
+            br.readLine();
+        } catch (IOException e) {
+            throw new CommandException(MESSAGE_IMPORT_VOLUNTEER_FAILED);
+        }
+
+        try {
+            String currLine;
+            while ((currLine = br.readLine()) != null) {
+                String[] arrayLine = currLine.split(",");
+                String args = "";
+                for (String i : arrayLine) {
+                    args += i + " ";
+                }
+                args = args.trim();
+                ArgumentMultimap argMultimap =
+                        ArgumentTokenizer.tokenize(args, PREFIX_NAME, PREFIX_GENDER,
+                                PREFIX_BIRTHDAY, PREFIX_PHONE, PREFIX_EMAIL,
+                                PREFIX_ADDRESS, PREFIX_TAG);
+
+                if (!arePrefixesPresent(argMultimap, PREFIX_NAME, PREFIX_GENDER, PREFIX_BIRTHDAY,
+                        PREFIX_ADDRESS, PREFIX_PHONE, PREFIX_EMAIL)
+                        || !argMultimap.getPreamble().isEmpty()) {
+                    throw new ParseException(String.format(MESSAGE_INVALID_COMMAND_FORMAT, AddCommand.MESSAGE_USAGE));
+                }
+
+                Name name = ParserUtil.parseName(argMultimap.getValue(PREFIX_NAME).get());
+                Gender gender = ParserUtil.parseGender(argMultimap.getValue(PREFIX_GENDER).get());
+                Birthday birthday = ParserUtil.parseBirthday(argMultimap.getValue(PREFIX_BIRTHDAY).get());
+                Phone phone = ParserUtil.parsePhone(argMultimap.getValue(PREFIX_PHONE).get());
+                Email email = ParserUtil.parseEmail(argMultimap.getValue(PREFIX_EMAIL).get());
+                Address address = ParserUtil.parseAddress(
+                        argMultimap.getValue(PREFIX_ADDRESS).get());
+                Set<Tag> tagList = ParserUtil.parseTags(argMultimap.getAllValues(PREFIX_TAG));
+
+                Volunteer volunteer = new Volunteer(name, gender, birthday, phone, email, address, tagList);
+
+                if (!model.hasVolunteer(volunteer)) {
+                    model.addVolunteer(volunteer);
+                    model.commitAddressBook();
```

Comment:
Why is the commit here? Placing it here means:
- When user uses the undo command, it only removes the add of the last volunteer added.
It is better to put the model.commitAddressBook() method outside. -> When ALL the volunteers have been added.

Reply:
noted and will update accordingly